### PR TITLE
LPS-188056 Generate OpenAPI from an API Application definition (with GET endpoints and returning a Page of items)

### PR DESCRIPTION
**Pull request by ccorreagg**
Related issue: https://liferay.atlassian.net/browse/LPS-188056

---

The purpose of this PR is to provide a mechanism to generate the OpenAPI specification file of an API Application definition. The only method allowed in the API Application definition by now is `GET` and the operation that will be performed is a `get<Entity>sPage`, that means the returned entity will be a Page of entities, and each entity is the one defined in the API Application.

For that, a new implementation of the `OpenAPIContributor` interface has been implemented. In the short future, it will be possible to get the OpenAPI by just calling the OpenAPI JAX-RS resource, but by now that functionality is being implemented in parallel.

---

**Data types supported:**

- Aggregation
- Attachment
- Boolean
- Date
- Date Time
- Decimal
- Integer
- Long Integer
- Long Text
- Multiselect Picklist
- Picklist
- Precision Decimal
- Rich Text
- Text

---

**How to test it:**

Deploy the following modules:
- `modules/apps/portal-vulcan/portal-vulcan-api`
- `modules/apps/portal-vulcan/portal-vulcan-impl`
- `modules/apps/headless/headless-builder/headless-builder-api`
- `modules/apps/headless/headless-builder/headless-builder-impl`

Go to the directory `modules/apps/headless/headless-builder/headless-builder-test` and execute `gw tI --tests APIApplicationOpenAPIContributorTest`

### CODE REVIEW

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/openapi/contributor/test/APIApplicationOpenAPIContributorTest.java [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+
+		_apiApplicationOpenAPIContributor.contribute(
+			openAPI,
+			new OpenAPIContext() {
+				{
+					setPath("/o/test/");
+				}
+			});
+
+		JSONUtil.equals(
```
Comment (LuismiBarcos) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/openapi/contributor/test/APIApplicationOpenAPIContributorTest.java:498]: Maybe I am missing something but shouldn't it have an assert? I don't see any `assert` on the test 😅

**Review by LuismiBarcos — COMMENTED** [review]
LGTM :) 👏 👏 👏 
Just a question

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/openapi/contributor/test/APIApplicationOpenAPIContributorTest.java (hunk 1 above)
Comment (ccorreagg) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/openapi/contributor/test/APIApplicationOpenAPIContributorTest.java:498]: 😂 you are totally right

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:stop

**Comment by ccorreagg** [comment]
Rebased with master. Will launch the tests again

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]

**Comment by ccorreagg** [comment]
Forwrded manually: https://github.com/brianchandotcom/liferay-portal/pull/136681

**Pull request closed without merging**